### PR TITLE
Cleaned up project-card.js by using lodash template

### LINKED ISSUE #24: Add template engine
Instead of building html straight from JavaScript by combining a bunch of variables and strings together, e.g.,

```js
var html = '<div><h1>' + header + '</h1><p>' + paragraphContent + '<p></div>'
```

Let's add a template engine to keep the codebase cleaner.

### PR DESCRIPTION
**Pull request by mmmavis**
Fixes #24

### CODE REVIEW

**Comment by xmatthewx** [comment]
ohh. very nice. sorry that i never got around to a template. what do you think of lodash? looked like a very active [update to underscore](http://stackoverflow.com/questions/13789618/differences-between-lodash-and-underscore).

**Comment by mmmavis** [comment]
@acabunoc / @cadecairos r? 😉

**Comment by mmmavis** [comment]
@xmatthewx oh no worries at all! I've never used lodash before but yea as you mentioned it seems to be a more actively updated version of underscore. So far every method I used is just the same as its underscore counterpart.

**Comment by xmatthewx** [comment]
@acabunoc / @cadecairos r?

**Comment by mmmavis** [comment]
hey @cadecairos / @acabunoc, I know you two are having a busy week :wink: . If time is too limited I'd like you to review the other 2 open PRs instead. This PR is a pretty straightforward switchover. Changes are pretty tedious so it might take long to review. Let's just merge this PR - it shouldn't break anything. Agree? :smile:

**Comment by xmatthewx** [comment]
@mmmavis - we might create a bug while Abby chats with program people. Maybe another engineer has a moment to spare? If I can find time, I can test and QA.

**Comment by cadecairos** [comment]
@mmmavis sorry for not being super responsive, I'll prioritize your PRs for pulse this afternoon

File: index.html [hunk 1]
```diff
@@ -54,6 +54,38 @@ <h4 id="starredHeader" class="hidden">Starred</h4>
     <p>Brought to you by the Network. <a href="https://github.com/xmatthewx/network-pulse">Contribute</a> to this prototype.</p>
   </footer>
 
+  <!-- Project Card template -->
+  <script type="text/template" id="project-card">
+    <%
+      var classNames = [
+        "project",
+        typeof(colorName)!== 'undefined' ? colorName : "",
```
Comment (mmmavis) [inline index.html]: had to do`typeof` here to check the `var`... simply checking `colorName` would break `lodash`

Comment (cadecairos) [inline index.html]: Just curious, break it in what way?

**Comment by cadecairos** [comment]
Looks great, R+

**Comment by cadecairos** [comment]
I have to pull back my R+

**Comment by cadecairos** [comment]
If you set any field displayed in the template to something like:

```
<img src=x onerror=alert('pwned')></img>
```

It will execute the javascript!

We need to either figure out if lodash can automatically sanitize inputs to the templates, or grab something to do it for us (writing sanitizers from scratch is never a good idea)

File: index.html (hunk 1 above)
Comment (mmmavis) [inline index.html]: `lodash` will not work if we are trying to check for an `undefined` var, e.g., `if (notExistVar)` ---> `lodash` stops working

**Comment by xmatthewx** [comment]
Could someone else change the inputs? If so, will we need scrap lodash?

**Comment by mmmavis** [comment]
Let me see if I can find a simple solution to this, if not... can we create a follow up ticket for this - Either to find another template engine alternative OR find a way to sanitize inputs. Even without `lodash` we still have the same issue since jQuery doesn't sanitize inputs either. So yea this is not a `lodash` problem but the fact we forgot to take sanitizers into consideration. Thanks @cadecairos for pointing that out! 👍

**Comment by mmmavis** [comment]
https://lodash.com/docs#template

`<%- someVar %>` instead of `<%= someVar %>` solves the problem! yay!

**Comment by cadecairos** [comment]
Yessss, nice find!

**Comment by mmmavis** [comment]
@cadecairos R? 😉

**Comment by cadecairos** [comment]
R+

**Comment by mmmavis** [comment]
yayyaya Thanks for much @cadecairos !

**Pull request merged**